tests/test_clock.py

```diff
@@ -30,3 +32,57 @@ def test_negative_numbers_raise_ValueError(self):
     def test_now_is_timezone_aware(self):
         """now() should always return a tz-aware datetime."""
         self.assertIsNotNone(self.clock.now().tzinfo)
+
+
+class TimerTest(unittest.TestCase):
+
+    def setUp(self):
+        self.mock_clock = mock.Mock()
+
+    def test_timer_duration_exceeded(self):
+        duration = datetime.timedelta(hours=3 * 24)
+        self.mock_clock.now.side_effect = [
+            datetime.datetime(
+                2016, 7, 23, 10, 51, 9, 928000, tzinfo=pytz.utc),
+            datetime.datetime(
+                2016, 7, 27, 10, 51, 9, 928000, tzinfo=pytz.utc)
+        ]
+        self.assertTrue(clock.Timer(self.mock_clock, duration).expired())
+
+    def test_timer_duration_matched(self):
+        duration = datetime.timedelta(hours=3 * 24)
+        self.mock_clock.now.side_effect = [
+            datetime.datetime(
+                2016, 7, 23, 10, 51, 9, 928000, tzinfo=pytz.utc),
+            datetime.datetime(
+                2016, 7, 26, 10, 51, 9, 928000, tzinfo=pytz.utc)
+        ]
+        self.assertFalse(clock.Timer(self.mock_clock, duration).expired())
+
+    def test_timer_duration_not_matched_or_exceeded(self):
+        duration = datetime.timedelta(hours=3 * 24)
+        self.mock_clock.now.side_effect = [
+            datetime.datetime(
+                2016, 7, 23, 10, 51, 9, 928000, tzinfo=pytz.utc),
+            datetime.datetime(
+                2016, 7, 24, 10, 51, 9, 928000, tzinfo=pytz.utc)
+        ]
+        self.assertFalse(clock.Timer(self.mock_clock, duration).expired())
+
+    def test_reset(self):
+        self.mock_clock.now.side_effect = [
+            datetime.datetime(
+                2016, 7, 23, 10, 51, 9, 928000, tzinfo=pytz.utc),
+            datetime.datetime(
+                2016, 7, 26, 10, 51, 9, 928000, tzinfo=pytz.utc)
+        ]
+        timer = clock.Timer(self.mock_clock, None)
+        self.assertEqual(
+            timer._start_time,
+            datetime.datetime(
+                2016, 7, 23, 10, 51, 9, 928000, tzinfo=pytz.utc))
+        timer.reset()
+        self.assertEqual(
+            timer._start_time,
+            datetime.datetime(
+                2016, 7, 26, 10, 51, 9, 928000, tzinfo=pytz.utc))
```

Comment:
Peeking into timer's internal variables is a bit ugly. Tests are generally more robust if they call the system under test the same way that a real client would. But this one is a bit awkward to test because it's hard to verify from the outside that expire actually reset the clock correctly.

I feel like it's sufficient to do something like:

1. Force timer to expiration
1. Reset
1. Assert that timer is no longer expired
1. (maybe?) Verify we can expire it again